content/en/docs/develop/core-modules/bank.md

```diff
@@ -6,23 +6,94 @@ weight: 40
 {{< alert >}}
 **Note**
 
-XPLA Chain's bank module inherits from the Cosmos SDK's [`bank`](https://docs.cosmos.network/master/modules/bank/) module. This document is a stub and covers mainly important XPLA Chain-specific notes about how it is used.
+XPLA Chain's bank module inherits from the Cosmos SDK's [`bank`](https://docs.cosmos.network/v0.45/modules/bank/) module. This document is a stub and covers mainly important XPLA Chain-specific notes about how it is used.
 {{< /alert >}}
 
 The bank module is the base transactional layer of the XPLA Chain. This module allows assets to be sent from one `Account` to another. The bank module defines the following types of send transactions: `MsgSend` and `MsgMultiSend`.
 
+The bank module is responsible for handling multi-asset coin transfers between accounts and tracking special-case pseudo-transfers which must work differently with particular kinds of accounts (notably delegating/undelegating for vesting accounts). It exposes several interfaces with varying capabilities for secure interaction with other modules which must alter user balances.
+
+In addition, the bank module tracks and provides query support for the total supply of all assets used in the application.
+
+This module will be used in the XPLA Chain.
+
+### Supply
+
+The `supply` functionality:
+
+- passively tracks the total supply of coins within a chain,
+- provides a pattern for modules to hold/interact with `Coins`, and
+- introduces the invariant check to verify a chain's total supply.
+
+#### Total Supply
+
+The total `Supply` of the network is equal to the sum of all coins from the
+account. The total supply is updated every time a `Coin` is minted (eg: as part
+of the inflation mechanism) or burned (eg: due to slashing or if a governance
+proposal is vetoed).
+
+### Module Accounts
+
+The supply functionality introduces a new type of `auth.Account` which can be used by
+modules to allocate tokens and in special cases mint or burn tokens. At a base
+level these module accounts are capable of sending/receiving tokens to and from
+`auth.Account`s and other module accounts. This design replaces previous
+alternative designs where, to hold tokens, modules would burn the incoming
+tokens from the sender account, and then track those tokens internally. Later,
+in order to send tokens, the module would need to effectively mint tokens
+within a destination account. The new design removes duplicate logic between
+modules to perform this accounting.
+
+The `ModuleAccount` interface is defined as follows:
+
+```go
+type ModuleAccount interface {
+  auth.Account               // same methods as the Account interface
+
+  GetName() string           // name of the module; used to obtain the address
+  GetPermissions() []string  // permissions of module account
+  HasPermission(string) bool
+}
+```
+
+> **WARNING!**
```

Comment:
Nit: suggest using this format `{{< alert context="warning" >}}`

Reply:
https://github.com/xpladev/docs/pull/13/commits/f9fe3f2d179d843f6fc7af01d600ecbdb7f50680